### DISCUSSION TITLE
Is there any way to fix transport UMAP?

### DISCUSSION THREAD

**Discussion by jjoon0928** [lmcinnes/umap#1086] [Q&A] [answered]
Hello, I am really impressed by your visualization method.

I am currently studying Deep metric learning, and there are many papers which use UMAP as visualization method for deep metric learning. In particular, UMAP is usually used for Hyperbolic metric learning on Poincare disk.  When I use UMAP, I could check the data, proxies and their relations were visualized well. However, I would like to observe the process of training. But it is not natural. I mean that the result is visualized well but I would like to visualise how the proxies are being updated. 

Proxies are generated by the code like below:
torch.randn(512, 512)  refers to 512 number of proxies with 512 size of embeddings.

What I would like to know is how can I consistently reduce those dimension to 2-dimension to check the proxies are updated.

**Comment by lmcinnes** [comment, marked as answer]
I'm not sure I fully understand the question, but I think AlignedUMAP might be the right approach for this, as it should keep the embeddings aligned reasonably.

**Reply by jjoon0928** [reply]
Thank you for your comment!

AlignedUMAP also provides Hyperboloid and Poincare metric?
